# HG changeset patch
# User Stefan Hepp <stefan@stefant.org>
# Date 1259046120 -3600
#      Tue Nov 24 08:02:00 2009 +0100
# Branch release-3-2-x
# Node ID 52d8ad5d28d6925ee65331f03344928d89ccbb23
# Parent  ae70ce64beb7968ea0417dab36a62dddd0941ca3
gnuplot_drawnow.m (gnuplot_default_term): don't set term to x11 unless DISPLAY is set

diff --git a/doc/interpreter/contributors.in b/doc/interpreter/contributors.in
--- a/doc/interpreter/contributors.in
+++ b/doc/interpreter/contributors.in
@@ -67,16 +67,17 @@ Kai Habel
 William P. Y. Hadisoeseno
 Jaroslav Hajek
 Benjamin Hall
 Kim Hansen
 Soren Hauberg
 Dave Hawthorne
 Daniel Heiserer
 Martin Helm
+Stefan Hepp
 Yozo Hida
 Ryan Hinton
 Roman Hodek
 A. Scottedward Hodel
 Richard Allan Holcombe
 Tom Holroyd
 David Hoover
 Kurt Hornik
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-11-02  Stefan Hepp  <stefan@stefant.org>
+
+	* plot/gnuplot_drawnow.m (gnuplot_default_term): Don't set term to
+	x11 unless DISPLAY is set.
+
 2009-11-11  John W. Eaton  <jwe@octave.org>
 
 	* plot/stairs.m (__stairs__): Correct nargin check.  New demos.
 	From Jakub Kasse <jakub.kasse@tul.cz>.
 
 2009-09-01  Christophe Tournery  <christophe.tournery@illusonic.com>
 
 	* audio/wavread.m: Rename data_size from obsolete ck_size.
diff --git a/scripts/plot/gnuplot_drawnow.m b/scripts/plot/gnuplot_drawnow.m
--- a/scripts/plot/gnuplot_drawnow.m
+++ b/scripts/plot/gnuplot_drawnow.m
@@ -321,18 +321,20 @@ endfunction
 function term = gnuplot_default_term ()
   term = getenv ("GNUTERM");
   ## If not specified, guess the terminal type.
   if (isempty (term))
     if (ismac ())
       term = "aqua";
     elseif (! isunix ())
       term = "windows";
+    elseif (! isempty (getenv ("DISPLAY")))
+      term = "x11";
     else
-      term = "x11";
+      term = "unknown";
     endif
   endif
 endfunction
 
 function [term, opts] = gnuplot_trim_term (string)
   ## Extract the terminal type and terminal options (from print.m)
   string = deblank (string);
   n = strfind (string, ' ');
